components/edit_channel_purpose_modal/edit_channel_purpose_modal.tsx

```diff
@@ -102,24 +91,23 @@ class EditChannelPurposeModal extends React.PureComponent {
         }
 
         this.setState({requestStarted: true});
-        const {data, error} = await patchChannel(channel.id, {purpose});
+        const result = await patchChannel(channel.id, {...channel, purpose});
         this.setState({requestStarted: false});
-
-        if (data) {
-            this.unsetError();
-            this.onHide();
-        } else if (error) {
-            this.setError(error);
+        if ('error' in result) {
+            this.setError(result.error);
+            return;
         }
+        this.unsetError();
+        this.onHide();
     }
 
-    handleChange = (e) => {
+    handleChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
         e.preventDefault();
         this.setState({purpose: e.target.value});
     }
 
-    getPurpose = (node) => {
-        this.purpose = node;
+    getPurpose = (node: HTMLTextAreaElement) => {
+        this.setState({...this.state, purpose: node.value});
```

Comment:
`this.purpose` is the ref for the `textarea` used below, whereas `this.state.purpose` is a field storing the purpose text. We'll want to unmerge these values here and everywhere else this has been done.

Reply:
This should be reverted to the original state, with a new `purpose` ref that is assigned to the `textarea` below being used here and in `handleEntering`.